be/src/formats/parquet/file_reader.cpp

```diff
@@ -233,15 +235,16 @@ void FileReader::_build_split_tasks() {
         split_ctx->split_start = start_offset;
         split_ctx->split_end = end_offset;
         split_ctx->file_metadata = _file_metadata;
-        _scanner_ctx->split_tasks->emplace_back(std::move(split_ctx));
+        _scanner_ctx->split_tasks.emplace_back(std::move(split_ctx));
```

Comment:
I'm afraid if we don't filter unneeded row groups here, we might generate some split simply do nothing.

Reply:
I agree. It's better to put filter code to filter split task early.

And if that, then in the first comment, split tasks are not consecutive any more. And my code can still handle inconsecutive split tasks.

Reply:
And I think we can do that in parquet, because we have never met parquet group size is too small.

However I suggest we'd better not do that in orc, because if we do early filter, myabe some tiny stripes are inconsecutive any more, and maybe it will incur more io reads.